api/common/crypto.cc

```diff
@@ -98,8 +98,7 @@ bool EncAes256CbcSha512(const std::string &key,
 
     // Build last block content with padding
     memcpy(last_block, clear_last_block, clear.length() % AES_BLOCK_SIZE);
-    memset(last_block + AES_BLOCK_SIZE - pad_length, pad_length,
-           AES_BLOCK_SIZE);
+    memset(last_block + AES_BLOCK_SIZE - pad_length, pad_length, pad_length);
```

Comment:
I'm not sure about this one:
first the 2nd argument should be a character, here it's a length (which happen to he a character too, but a little random)
2: I think the memset shoud set what hasn't been set by memcpy first, not overrite data
so maybe 3rd argument should be (pad_length - clear.length() % AES_BLOCK_SIZE)